packages/shared-core/src/ids.ts

```diff
@@ -0,0 +1,74 @@
+import { SEPARATOR, DocumentType, VirtualDocumentType } from "@budibase/types"
+
+// Because DocumentTypes can overlap, we need to make sure that we search
+// longest first to ensure we get the correct type.
+const sortedDocumentTypes = Object.values(DocumentType).sort(
+  (a, b) => b.length - a.length // descending
+)
+
+export function getDocumentType(id: string): DocumentType | undefined {
+  return sortedDocumentTypes.find(docType =>
+    id.startsWith(`${docType}${SEPARATOR}`)
+  )
+}
+
+// Because VirtualDocumentTypes can overlap, we need to make sure that we search
+// longest first to ensure we get the correct type.
+const sortedVirtualDocumentTypes = Object.values(VirtualDocumentType).sort(
+  (a, b) => b.length - a.length // descending
+)
+export function getVirtualDocumentType(
+  id: string
+): VirtualDocumentType | undefined {
+  return sortedVirtualDocumentTypes.find(docType =>
+    id.startsWith(`${docType}${SEPARATOR}`)
+  )
+}
+
+const EXTERNAL_TABLE_ID_REGEX = new RegExp(
+  `^${DocumentType.DATASOURCE_PLUS}_(.+)__(.+)$`
+)
+
+const idCheckFor = (
+  type: DocumentType | VirtualDocumentType
+): ((id: string) => boolean) => {
+  return (id: string): boolean => !!id && id.startsWith(`${type}${SEPARATOR}`)
+}
+
+export const isExternalTableId = (id: string): boolean => {
+  const matches = id.match(EXTERNAL_TABLE_ID_REGEX)
+  return !!id && matches !== null
+}
+
+/**
+ *  This function checks if the given ID is a table ID, including datasource
+ *  plus tables.
+ *
+ *  Example of IDs that will return true:
+ *  - "ta_12345"
+ *  - "datasource_plus_12345__ta_67890"
+ */
+export const isTableIdOrExternalTableId = (id: string): boolean => {
```

Comment:
This used to be called `isTableId` but that felt potentially dangerous to me so I've given it a more explicit name.

Same for `isDatasourceOrDatasourcePlusId`.